test/e2e/storage/persistent_volumes-local.go

```diff
@@ -53,98 +60,309 @@ const (
+			Expect(pod2Err).NotTo(HaveOccurred())
+
+			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(config.client, pod2))
+			pod2NodeName, pod2NodeNameErr := podNodeName(config, pod2)
+			Expect(pod2NodeNameErr).NotTo(HaveOccurred())
+			framework.Logf("Pod2 %q created on Node %q", pod2.Name, pod2NodeName)
+			Expect(pod2NodeName).To(Equal(node0.Name))
+
+			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
+
+			By("Writing in pod1")
+			podRWCmdExec(pod1, writeCmd)
+			By("Reading in pod2")
+			readOut := podRWCmdExec(pod2, readCmd)
+
+			Expect(readOut).To(ContainSubstring(testVol.hostDir)) /*aka writeTestFileContents*/
+
+			By("Deleting pod1")
+			framework.DeletePodOrFail(config.client, config.ns, pod1.Name)
+			By("Deleting pod2")
+			framework.DeletePodOrFail(config.client, config.ns, pod2.Name)
+		})
+	})
+
+	Context("when using local volume provisioner", func() {
+		var (
+			volumePath string
+		)
+
+		BeforeEach(func() {
+			setupLocalVolumeProvisioner(config, node0)
+			volumePath = path.Join(
+				hostBase, discoveryDir, fmt.Sprintf("vol-%v", string(uuid.NewUUID())))
+		})
+
+		AfterEach(func() {
+			cleanupLocalVolumeProvisioner(config, node0, volumePath)
+		})
+
+		It("should create and recreate local persistent volume", func() {
+			By("Creating bootstrapper pod to start provisioner daemonset")
+			createBootstrapperPod(config)
+			kind := schema.GroupKind{"extensions", "DaemonSet"}
+			framework.WaitForControlledPodsRunning(config.client, config.ns, daemonSetName, kind)
+
+			By("Creating a directory under discovery path")
+			mkdirCmd := fmt.Sprintf("mkdir %v", volumePath)
+			runLocalUtil(config, node0.Name, mkdirCmd, hostBase)
+			err := waitForLocalPersistentVolume(config.client, volumePath)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create a persistent volume claim for local volume: the above volume will be bound.
+			// Note: make sure no other local PVs are available; otherwise, the test can be flaky.
+			By("Creating a persistent volume claim")
+			claim, err := config.client.Core().PersistentVolumeClaims(config.ns).Create(newLocalClaim(config))
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitForPersistentVolumeClaimPhase(
+				v1.ClaimBound, config.client, claim.Namespace, claim.Name, framework.Poll, 1*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Delete the persistent volume claim: file will be cleaned up and volume be re-created.
+			By("Deleting the persistent volume claim to clean up persistent volume and re-create one")
+			writeCmd, readCmd := createWriteAndReadCmds(volumePath, testFile, testFileContent)
+			runLocalUtil(config, node0.Name, writeCmd, hostBase)
+			err = config.client.Core().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, &metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			err = waitForLocalPersistentVolume(config.client, volumePath)
```

Comment:
Can you compare the UUID of the pv objects to make sure that a new object was created?